cli/src/main/kotlin/uk/gov/justice/digital/client/BlockingDomainClient.kt

```diff
@@ -22,32 +29,61 @@ interface DomainClient {
 @Singleton
 class BlockingDomainClient : DomainClient {
 
-    @field:Client("\${micronaut.http.services.domain.url}")
+    @Inject
+    private lateinit var objectMapper: ObjectMapper
     @Inject
     private lateinit var client: HttpClient
 
+    @Value("\${http.client.url}")
+    private lateinit var baseUrl: String
+
+    private val DOMAIN_RESOURCE by lazy {
+        UriBuilder.of("$baseUrl/domain").build()
+    }
+
     override fun getDomains(): Array<Domain> = client.get<Array<Domain>>(DOMAIN_RESOURCE)
 
     override fun getDomainWithName(name: String): Domain? {
         val requestUri =
             UriBuilder.of(DOMAIN_RESOURCE)
                 .queryParam("name", name)
                 .build()
-                .toASCIIString()
 
         return client.get<Array<Domain>>(requestUri).firstOrNull()
     }
 
-    private inline fun <reified T> HttpClient.get(url: String): T =
-        this.toBlocking().retrieve(
-            GET<String>(url)
-                .accept(APPLICATION_JSON)
-                .header(USER_AGENT, "domain-builder-cli/v0.0.1"),
-            Argument.of(T::class.java)
-        )
+    private inline fun <reified T> HttpClient.get(url: URI): T {
+        val request = HttpRequest.newBuilder()
+            .GET()
+            .headers(
+                ACCEPT, APPLICATION_JSON,
+                USER_AGENT, "domain-builder-cli/v0.0.1"
+            )
+            .uri(url)
+            .timeout(REQUEST_TIMEOUT)
+            .build()
+
+        val response = client.send(request, HttpResponse.BodyHandlers.ofInputStream())
+
+        if (response.statusCode() in 200..299)
+            return objectMapper.readValue(response.body(), T::class.java) ?: throw RuntimeException("No data in response")
+        else throw RuntimeException("Server returned an error")
```

Comment:
do we need a finally block to close the client connection?

Reply:
This isn't necessary.